art/metrics/privacy/membership_leakage.py

```diff
@@ -109,19 +124,113 @@ def PDTP(  # pylint: disable=C0103
             alt_pred_bin_indexes = np.digitize(alt_pred, bins)
             alt_pred_bin_indexes[alt_pred_bin_indexes == 101] = 100
             alt_pred_bin = bins[alt_pred_bin_indexes] - 0.005
-            ratio_1 = pred_bin / alt_pred_bin
-            ratio_2 = alt_pred_bin / pred_bin
-            # get max value
-            max_value = max(ratio_1.max(), ratio_2.max())
+            if comparison_type == ComparisonType.RATIO:
+                ratio_1 = pred_bin / alt_pred_bin
+                ratio_2 = alt_pred_bin / pred_bin
+                # get max value
+                max_value = max(ratio_1.max(), ratio_2.max())
+            elif comparison_type == ComparisonType.DIFFERENCE:
+                max_value = np.max(abs(pred_bin - alt_pred_bin))
+            else:
+                raise ValueError("Unsupported comparison type.")
             iter_results.append(max_value)
         results.append(iter_results)
 
     # get average of iterations for each sample
-    # We now have a list of list, internal lists represent an iteration. We need to transpose and get averages.
+    # We now have a list of lists, internal lists represent an iteration. We need to transpose and get averages.
     per_sample = list(map(list, zip(*results)))
     avg_per_sample = np.array([sum(val) / len(val) for val in per_sample])
     worse_per_sample = np.max(per_sample, axis=1)
     std_dev_per_sample = np.std(per_sample, axis=1)
 
     # return avg+worse leakage + standard deviation per sample
     return avg_per_sample, worse_per_sample, std_dev_per_sample
+
+
+def SHAPr(  # pylint: disable=C0103
+    target_estimator: "Classifier",
```

Comment:
Can we use here `CLASSIFIER_TYPE` from `art.utils`?
```suggestion
    target_estimator: CLASSIFIER_TYPE,
```